src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -544,6 +544,33 @@ void Simulator::poll_topics()
 	if (updated) {
 		orb_copy(ORB_ID(vehicle_status), _vehicle_status_sub, &_vehicle_status);
 	}
+
+	orb_check(_tune_control_sub, &updated);
+
+	if (updated) {
+
+		orb_copy(ORB_ID(tune_control), _tune_control_sub, &_tune_control);
+
+		_tunes.set_control(_tune_control);
+
+		mavlink_play_tune_t tune_msg = { };
+		mavlink_message_t message = { };
+
+		union {
+			struct {
+				unsigned int frequency;
+				unsigned int duration;
+				unsigned int silence_length;
```

Comment:
@catch-twenty-two , if you change `silence_length` to `silence` you will match the naming convention of the `tune_control_s` type.  (You could consider utilizing that type to store these values, although it could require modifying how you are using the union's memory space to transfer values to the message.)

Reply:
I ended up removing the union, I didn't really need it any more.  I still create a local structure  just to be on the safe side in case the tune_control_s structure changes, and I think it makes the code a little more readable.